africanus/gridding/simple/gridding.py

```diff
@@ -43,142 +54,96 @@ def _grid(vis, uvw, flags, ref_wave, nx, ny, convolution_filter):
     # one grid for the resampled visibilities per correlation:
     grid = np.zeros((vis.shape[2],ny,nx),dtype=np.complex64)
 
-    # 1. Foreach row in block
-    #    (A) apply beam DDE gains
-    #        Sample the facet centre position
-    #        in the beam cube as it varies over time
-    #        to produce these gains.
-    #    (B) average
-    # 2. Foreach row in averaged block
+    # for deconvolution the PSF should be 2x size of the image (see
+    # Hogbom CLEAN for details), one grid for the sampling function:
+    # Reference the real component of the PSF, imaginary is zero
+    psf = np.zeros((2*ny,2*nx), dtype=np.complex64)
```

Comment:
True, ideally you need a PSF that is twice the size of the image but we often want speed things up by using CLEAN boxes (known as Clark CLEAN). You might want to add an input parameter which specifies how large the PSF should be (as a fraction of the image size for example). Also, when running multiple major cycles, you do not want to recompute the PSF every time. So you should also add a parameter that specifies whether the PSF should be computed or not. Having said that, I don't see the weight column being passed in. Are we currently assuming that all visibilities are equally weighted?